lib/sns-trigger.js

```diff
@@ -204,7 +311,44 @@ module.exports = class SnsTrigger {
 			});
 		}
 
-		return hasAttributes && parsedAttributes;
+		parsedAttributes.topicName = {
+			DataType: 'String',
+			StringValue: topicName
+		};
+
+		return parsedAttributes;
+	}
+
+	parseExtraProperties(event, sqsName) {
+
+		if(!this.session?.clientCode)
+			throw new SnsTriggerError('The session must have a clientCode', SnsTriggerError.codes.MISSING_CLIENT_CODE);
+
+		return {
+			...event.payloadFixedProperties && { payloadFixedProperties: event.payloadFixedProperties },
+			contentS3Path: this.getContentS3Path(sqsName)
+		};
+	}
+
+	getTopicNameFromArn(snsArn) {
+
+		const isValidSqsUrl = this.isValidSnsArn(snsArn);
+
+		if(!isValidSqsUrl)
+			throw new SnsTriggerError(`Invalid SNS ARN: ${snsArn}`, SnsTriggerError.codes.INVALID_SNS_ARN);
+
+		const arnParts = snsArn.split(':');
+		let snsTopic = arnParts[arnParts.length - 1];
+
+		if(snsTopic.search('fifo') !== -1)
+			[snsTopic] = snsTopic.split('.');
+
+		return snsTopic;
```

Comment:
Probablemente sea mas performante y seguro (acá estas matcheando `fifo` en cualquier parte del nombre), usar `.endsWith()`...

```suggestion
		const snsTopic = arnParts[arnParts.length - 1];

		if(snsTopic.endsWith('.fifo'))
			return snsTopic.substring(0, snsTopic.length - 5);

		return snsTopic;
```